libs/barista-components/filter-field/src/filter-field.ts

```diff
@@ -827,6 +840,216 @@ export class DtFilterField<T = any>
     }
   }
 
+  /** @internal Handles keyup on the input */
+  _handleInputKeyUp(event: KeyboardEvent): void {
+    const keyCode = _readKeyCode(event);
+    this._getCurrentFocusState();
+
+    if (
+      !this._tags.length ||
+      this._currentFocusState !== 'inputboundary' ||
+      keyCode !== LEFT_ARROW
+    ) {
+      return;
+    }
+
+    // The input element moves the cursor on keydown but the navigation with arrow keys is on keyup,
+    // causing the cursor to jump over position 0 in the input and focuses the first focusable tag element.
+    // For that, allowArrowKeyFocusOnTags retains getting to position 0 before focusing a tag element.
+    if (this._allowArrowKeyFocusOnTags || !this._inputValue.length) {
+      this._handleArrowKeyNavigation({ direction: 'left' });
+    } else {
+      this._allowArrowKeyFocusOnTags = this._getAllowArrowKeyFocusOnTags();
+    }
+  }
+
+  /** @internal Handles the navigation through the list of tags when using arrow keys */
+  _handleArrowKeyNavigation(navigationEvent: {
+    /** Contains the currently focused tag from which the event was fired from */
+    currentTag?: DtFilterFieldTag;
+    /** Whether to step back */
+    direction: 'left' | 'right';
+  }): void {
+    const { currentTag, direction } = navigationEvent;
+
+    // Step focus out of input element and into tag list and early exit
+    if (currentTag === undefined) {
+      this._getFirstTagFromInput(direction)?.nativeElement.focus();
+      return;
+    }
+
+    const focusedTagIndex = this._tags
+      .toArray()
+      .findIndex((tag) => tag.editButton === currentTag?.editButton);
+
+    this._getCurrentlyFocusedTagType(currentTag!);
+
+    if (direction === 'left') {
+      this._findPreviousTagDefaultMode(
+        currentTag!,
```

Comment:
Should not be necessary, as you check for undefined above already
```suggestion
        currentTag,
```